File: frontend/src/app/model/state.ts
import { Country } from "./country";


export class State {
    id: number;
    name: string;
    code: string;
    country: Country;

    constructor(id: number, name: string, code: string, country: Country) {
        this.id = id;
        this.name = name;
        this.code = code;
        this.country = country;
    }
}

File: backend/Restaurant-Project/src/main/java/com/morgan/restaurant/model/State.java
package com.morgan.restaurant.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@NoArgsConstructor
@Data
@AllArgsConstructor
@Entity
@Table(name = "state")
public class State {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "country_id")
    private Country country;
}
